src/utils/log.h

```diff
@@ -30,21 +30,67 @@
 #ifndef G_UTILS_LOG_H
 #define G_UTILS_LOG_H
 
+#include <cstdio>
+#include <string>
+#include <type_traits>
+#include <utility>
 
-namespace giada {
-namespace u {
-namespace log 
-{
-/* init
-Initializes logger. Mode defines where to write the output: LOG_MODE_STDOUT,
-LOG_MODE_FILE and LOG_MODE_MUTE. */
+#include "utils/fs.h"
+#include "core/const.h"
 
-int init(int mode);
+namespace giada::u {
 
-void close();
+struct log {
```

Comment:
Just curious, why a struct and not a regular function in the giada::u::log namespace?

Reply:
Rebooting my brain on this PR I completely forgot about and I was just pinged by https://github.com/monocasual/giada/pull/398#discussion_r511637899...
The new `print` use metaprogramming, so it has to be in the `.h`.
Unfortunately it needs access to the variables `f`, `mode` & `stat`, so they are moved from the `.cpp` into the `.h`.
Then we have the problem of their allocation and initialization which is solved with the use of C++17 `inline` variables.
Usually I use this for static class variables and I just used the the struct `log` here is as a namespace since the class is not instantiated.
But actually `inline` variables work without classes, so I switched back to namespace to avoid changing the code too much.